OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -194,17 +197,15 @@ void Deploy(bool init)
 
 			if (!string.IsNullOrEmpty(info.DeploySound))
 				Game.Sound.Play(info.DeploySound, self.CenterPosition);
-
+			if (!string.IsNullOrEmpty(info.DeployAnimation))
+				body.Value.PlayCustomAnimation(self, info.DeployAnimation);
 			// Revoke upgrades that are used while undeployed.
 			if (!init)
 				OnDeployStarted();
 
 			// If there is no animation to play just grant the upgrades that are used while deployed.
 			// Alternatively, play the deploy animation and then grant the upgrades.
-			if (string.IsNullOrEmpty(info.DeployAnimation) || body.Value == null)
-				OnDeployCompleted();
-			else
-				body.Value.PlayCustomAnimation(self, info.DeployAnimation, OnDeployCompleted);
+			OnDeployCompleted();
```

Comment:
This makes OnDeployCommand to run immediately, before the animation has completed.  That's not what we want to happen.